### PR TITLE
[Sprint maintain] Fix datetimepicker on android API 19

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Task: [Sprint maintain] Fix datetimepicker on android API 19
Link redmine: https://dev.framgia.com/redmine/issues/129813
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java [hunk 1]
```diff
@@ -243,73 +243,76 @@ public void onInputCompensationToError(String errorMessage) {
...
-            case LeaveTypeId.FORGOT_CARD_OUT:
-            case LeaveTypeId.FORGOT_TO_CHECK_OUT:
-            case LeaveTypeId.LEAVE_EARLY_WOMAN_A:
-            case LeaveTypeId.LEAVE_EARLY_WOMAN_M:
-                validateCheckoutTime(currentDateTime);
-                break;
-            case LeaveTypeId.IN_LATE_A:
-            case LeaveTypeId.IN_LATE_M:
-                if (mCurrentTimeSelected == CurrentTimeSelected.CHECK_IN) {
+        if (view.isShown()) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/requestleave/RequestLeaveViewModel.java]: change to
```java
if(!view.isShown()) {
return;
}
....
```

File: app/src/main/java/com/framgia/wsm/screen/requestovertime/RequestOvertimeViewModel.java [hunk 2]
```diff
@@ -139,21 +139,24 @@ public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
...
     @Override
     public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
-        String currentDateTime =
-                DateTimeUtils.convertDateTimeToString(mCurrentDate, hourOfDay, minute);
-        if (mIsFromTimeSelected) {
-            setToTime(null);
-            validateFromTime(currentDateTime);
-        } else {
-            validateToTime(currentDateTime);
+        if (view.isShown()) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/requestovertime/RequestOvertimeViewModel.java]: like above

File: app/src/main/java/com/framgia/wsm/widget/dialog/DialogManagerImpl.java [hunk 3]
```diff
@@ -240,9 +240,7 @@ public DialogManager dialogMonthYearPicker(DatePickerDialog.OnDateSetListener on
                 if (field.getName().equals(DATE_PICKER)) {
                     field.setAccessible(true);
                     mDatePicker = (DatePicker) field.get(mDatePickerDialog);
-                    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
-                        customDatePicker(mDatePicker);
-                    }
+                    customDatePicker(mDatePicker);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/widget/dialog/DialogManagerImpl.java]: Did you test on emulator have API < 21?

Comment (NguyenTienVinh) [inline app/src/main/java/com/framgia/wsm/widget/dialog/DialogManagerImpl.java]: Đã test trên API 19

**Pull request merged**